vm/systemSmartContracts/governance_test.go

```diff
@@ -1 +1,614 @@
+			_ = json.Unmarshal(value, genProposal)
+			require.Equal(t, gitHubCommit, genProposal.GitHubCommit)
+			require.Equal(t, startNonce, genProposal.StartVoteNonce)
+			require.Equal(t, stopNonce, genProposal.EndVoteNonce)
+			require.Equal(t, callerAddr, genProposal.IssuerAddress)
+			require.False(t, genProposal.Voted)
+
+			return
+		},
+	}
+
+	gsc, _ := NewGovernanceContract(args)
+	callInput := createVMInput(big.NewInt(100), "proposal", callerAddr, []byte("addr2"))
+	callInput.Arguments = [][]byte{
+		gitHubCommit,
+		[]byte(fmt.Sprintf("%d", startNonce)),
+		[]byte(fmt.Sprintf("%d", stopNonce)),
+	}
+
+	retCode := gsc.Execute(callInput)
+	require.Equal(t, vmcommon.Ok, retCode)
+}
+
+func TestGovernanceContract_ExecuteVoteYes(t *testing.T) {
+	testExecuteVote(t, []byte("yes"))
+}
+
+func TestGovernanceContract_ExecuteVoteNo(t *testing.T) {
+	testExecuteVote(t, []byte("no"))
+}
+
+func TestGovernanceContract_ExecuteVoteVeto(t *testing.T) {
+	testExecuteVote(t, []byte("veto"))
+}
+
+func TestGovernanceContract_ExecuteVoteDontCare(t *testing.T) {
+	testExecuteVote(t, []byte("dontCare"))
+}
+
+func testExecuteVote(t *testing.T, vote []byte) {
+	t.Parallel()
+
+	proposalToVote := []byte("proposalToVote")
+	validatorAddr := []byte("addr2")
+	callerAddr := []byte("addr1")
+	args := createMockGovernanceArgs()
+	args.Eei = &mock.SystemEIStub{
+		GetStorageCalled: func(key []byte) []byte {
+			if bytes.Equal(key, append([]byte(validatorPrefix), validatorAddr...)) {
+				validatorData := &ValidatorData{
+					Delegators: []*VoterData{
+						{Address: callerAddr, NumNodes: 1},
+					},
+				}
+
+				validatorDataBytes, _ := json.Marshal(validatorData)
+				return validatorDataBytes
+			}
+			generalProposal := &GeneralProposal{
+				Voted: true,
+			}
+			generalProposalBytes, _ := json.Marshal(generalProposal)
+			return generalProposalBytes
+		},
+		ExecuteOnDestContextCalled: func(destination, sender []byte, value *big.Int, input []byte) (output *vmcommon.VMOutput, err error) {
+			autionData := &AuctionData{
+				NumStaked: 1,
+			}
+			auctionDataBytes, _ := json.Marshal(autionData)
+
+			return &vmcommon.VMOutput{
+				ReturnCode: vmcommon.Ok,
+				ReturnData: [][]byte{auctionDataBytes},
+			}, nil
+		},
+	}
+
+	gsc, _ := NewGovernanceContract(args)
+	callInput := createVMInput(big.NewInt(0), "vote", callerAddr, []byte("addr2"))
+	callInput.Arguments = [][]byte{
+		proposalToVote,
+		vote,
+		validatorAddr,
+	}
+
+	retCode := gsc.Execute(callInput)
+	require.Equal(t, vmcommon.Ok, retCode)
+}
+
+// Test Scenario
+// A proposal is voted if it has 2 vote with yes
+// 1. Init governance smart contract
+// 2. WhiteList an address at genesis
+// 3. WhileList a new address, vote with 2 validators with yes and close white list proposal
+// 4. Create a new general proposal with new white listed address, vote with 2 validator with yes general proposal
+// and close proposal
+func TestGovernanceContract_ExecuteProposalCloseProposal(t *testing.T) {
```

Comment:
you might add a function for each step 1, 2, 3 and 4

Reply:
done.